src/auth/auth.service.ts

```diff
@@ -0,0 +1,73 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { ApiMessages } from 'src/common/constants';
+import { Env } from 'src/env';
+import axios from 'axios';
+import { GithubService } from 'src/github/github.service';
+import { TRequest } from 'src/common/types/user.types';
+import { RedisService } from '@liaoliaots/nestjs-redis';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private readonly jwtService: JwtService,
+    private readonly githubService: GithubService,
+    private readonly redisService: RedisService,
+  ) {}
+
+  async handleGitHubCallback(code: string) {
+    const clientId = Env.GITHUB.CLIENT_ID;
+    const clientSecret = Env.GITHUB.CLIENT_SECRET;
+
+    try {
```

Comment:
PrimathonAICodeReviewer

Consider adding validation for the `code` parameter in the `handleGitHubCallback` method to ensure it is not empty or null before making the API call.
```typescript
+ if (!code) {
+    throw new HttpException('Invalid code provided', HttpStatus.BAD_REQUEST);
+ }
```



<!-- This is an auto-generated comment by OSS PrimathonAICodeReviewer -->